homeassistant/components/elmax/binary_sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax sensor platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for zone in panel_status.zones:
+            entity = ElmaxSensor(
+                panel=coordinator.panel_entry,
+                elmax_device=zone,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            if entity.unique_id not in known_devices:
```

Comment:
You're now constructing an entity object to then throw it away 99% of the time. Can you check this prior to instantiating it ?